File: Backend/routes/task.py
from Models.Task import Task
from database import CRUD
from fastapi import APIRouter, HTTPException

task = APIRouter()

#Metodo para crear una tarea
@task.post('/api/tasks')
def save_task(task:Task):
    response = CRUD.create_task(task.dict())
    if response:
        return response
    raise HTTPException

#Metodo para obtener todas las tareas
@task.get('/api/tasks')
def get_all():
    response = CRUD.get_all_tasks()
    if response:
        return response
    raise HTTPException

#Metodo para obtener una tarea en especifico
@task.get('/api/tasks/{id}')
def get_task(id:str):
    task = CRUD.get_task(id)
    if task:
        return task
    raise HTTPException


#Metodo para eliminar una tarea
@task.delete('/api/tasks/{id}')
def delete_task(id:str):
    if CRUD.get_task(id) != "null":
        response = CRUD.delete_task(id)
        if response:
            return 'Eliminado correctamente'
    else:
        return 'No existe'
    raise HTTPException


#Metodo para modificar una tarea
@task.put('/api/tasks/{id}')
def update_task(id:str,task:Task):
    if CRUD.get_task(id) != "null":
        response = CRUD.update_task(id,task.dict())
        if response:
            return response
    else:
        return 'No existe'
    raise HTTPException

File: frontend/src/api/task.js
import axios from "axios";

const URL = import.meta.env.VITE_API || "http://localhost:8000";
const endpoint = URL + "/api/tasks";

export const fetchTask = (id) => axios.get(`${endpoint}/${id}`);

export const createTask = (newTask) => axios.post(endpoint, newTask);

export const updateTask = (id, task) => axios.put(`${endpoint}/${id}`, task);

export const deleteTask = (id) => axios.delete(`${endpoint}/${id}`);

export const fetchTasks = () => axios.get(endpoint);